packages/react-sdk/src/components/NoiseCancellation/NoiseCancellationProvider.tsx

```diff
@@ -76,10 +76,33 @@ export const NoiseCancellationProvider = (
   const hasCapability = useHasPermissions(
     OwnCapability.ENABLE_NOISE_CANCELLATION,
   );
-  const isSupported =
-    hasCapability &&
-    noiseCancellationAllowed &&
-    noiseCancellation.isSupported();
+  const [isSupportedByBrowser, setIsSupportedByBrowser] = useState<
+    boolean | undefined
+  >(undefined);
+
+  const [isSupported, setIsSupported] = useState<boolean | undefined>(
+    undefined,
+  );
+
+  useEffect(() => {
+    noiseCancellation
+      .isSupported()
+      .then((_isSupportedByBrowser) => {
+        setIsSupportedByBrowser(_isSupportedByBrowser);
+      })
+      .catch((err) =>
+        console.error(
+          `Can't determine if noise cancellation is supported`,
+          err,
+        ),
+      );
+  }, [noiseCancellation]);
+
+  useEffect(() => {
+    setIsSupported(
+      isSupportedByBrowser && hasCapability && noiseCancellationAllowed,
+    );
+  }, [isSupportedByBrowser, hasCapability, noiseCancellationAllowed]);
```

Comment:
`isSupported` doesn't have to be state in this case. It can be just

```
const isSupported = isSupportedByBrowser && hasCapability && noiseCancellationAllowed;
```

(see https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state for the explanation of this pattern)